### PR TITLE
Overload for CosmosDbStorage constructor

### PR DESCRIPTION
**Pull request by yanetc**

### CODE REVIEW

**Comment by yanetc** [comment]
Hi @imranmomin,
This is the correct way to do the PR?
Could you review it, please?
Thanks!

File: src/CosmosDbStorage.cs [hunk 1]
```diff
@@ -138,6 +152,24 @@ public static CosmosDbStorage Create(string url, string authSecret, string datab
...
+
+	/// <summary>
+	///     Creates and returns an instance of CosmosDbStorage
+	/// </summary>
+	///<param name="cosmosClient">An instance of CosmosClient</param>
+	/// <param name="databaseName">The name of the database to connect with</param>
+	/// <param name="containerName">The name of the collection/container on the database</param>
+	/// <param name="options">The CosmosClientOptions object to override any of the options</param>
+	/// <param name="storageOptions">The CosmosDbStorageOptions object to override any of the options</param>
+	public static CosmosDbStorage Create(CosmosClient cosmosClient, string databaseName, string containerName, CosmosDbStorageOptions storageOptions = null)
```
Comment (imranmomin) [inline src/CosmosDbStorage.cs:164]: Please implement overload for method `CreateAsync`

File: src/CosmosDbStorageExtensions.cs [hunk 2]
```diff
@@ -61,4 +61,24 @@ internal static class CosmosDbStorageExtensions
...
+	///     Enables to attach Azure Cosmos DB to Hangfire
+	/// </summary>
+	/// <param name="configuration">The IGlobalConfiguration object</param>
+	/// <param name="cosmosClient">An instance of CosmosClient</param>
+	/// <param name="database">The name of the database to connect with</param>
+	/// <param name="collection">The name of the collection on the database</param>
+	/// <param name="option"></param>
+	/// <param name="storageOptions">The CosmosDbStorage object to override any of the options</param>
+	/// <returns></returns>
+	public static IGlobalConfiguration<CosmosDbStorage> UseAzureCosmosDbStorage(this IGlobalConfiguration configuration, CosmosClient cosmosClient, string databaseName, string containerName, CosmosDbStorageOptions storageOptions = null)
```
Comment (imranmomin) [inline src/CosmosDbStorageExtensions.cs:75]: Please implement overload for method `UseAzureCosmosDbStorageAsync`

**Review by imranmomin — CHANGES_REQUESTED** [review]
Also can you create the test case of the new implementations. It should be under `CosmosDbStorageFacts`

File: src/CosmosDbStorage.cs [hunk 3]
```diff
@@ -138,6 +152,24 @@ public static CosmosDbStorage Create(string url, string authSecret, string datab
...
+	{
+		if (cosmosClient is null)
+			throw new ArgumentNullException(nameof(cosmosClient));
```
Comment (imranmomin) [inline src/CosmosDbStorage.cs:167]: Please move the null check inside the constructor

**Pull request closed without merging**